src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -293,20 +292,31 @@ private void OnSubjectBuffersConnected(object sender, SubjectBuffersConnectedEve
             }
         }
 
-        private void UpdateReferenceLocationsTask(JoinableTask<IInlineRenameLocationSet> findRenameLocationsTask)
+        private void UpdateReferenceLocationsTask()
         {
             _threadingContext.ThrowIfNotOnUIThread();
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
+
+            var currentOptions = _options;
+            var currentRenameLocationsTask = _allRenameLocationsTask;
+            var cancellationToken = _cancellationTokenSource.Token;
```

Comment:
mutable data accessed and stored on UI thread so that bg work being kicked off is certain to use the correct data.

Reply:
This is a good comment to add :)